java-checks/src/main/java/org/sonar/java/checks/helpers/UnitTestUtils.java

```diff
@@ -63,7 +63,8 @@
     // assertJ
     MethodMatchers.create().ofSubTypes("org.assertj.core.api.AbstractAssert").anyName().withAnyParameters().build(),
     // spring
-    MethodMatchers.create().ofTypes("org.springframework.test.web.servlet.ResultActions").names("andExpect").addParametersMatcher(t -> true).build(),
+    MethodMatchers.create().ofTypes("org.springframework.test.web.servlet.ResultActions").names("andExpect").withAnyParameters().build(),
+    MethodMatchers.create().ofTypes("org.springframework.test.web.servlet.ResultActions").names("andExpectAll").withAnyParameters().build(),
```

Comment:
When the type is the same, you can provide multiples names to the method `names`:
```suggestion
MethodMatchers.create().ofTypes("org.springframework.test.web.servlet.ResultActions").names("andExpect", "andExpectAll").withAnyParameters().build(),
```